### ISSUE TITLE
How does `irods-grid` interact with tree voting and resources marked down

### ISSUE THREAD

**Issue by kript** [irods/irods#6750]
## Feature

Can the interaction between `irods-grid` and `passthru` write/read weighting and `iadmin modresc XXX status down` be documented please?

I would like to know what to expect if I use `irods-grid pause` or `irods-grid shutdown` on a server which has a resource marked high weighting for read (or write) - is iRODS smart enough to have those resources vote 0 when a resource is marked stopped or paused? 

cheers

John

**Comment by alanking** [comment]
I'm going to use this setup to demonstrate some observations. Note that I didn't run the multi-replica scenario because I expect the result to be the same should the resource hosted on the paused/shutdown iRODS servers win the vote.
```
$ ilsresc
def:deferred
├── pt0:passthru
│   └── 110eb55edbccResource:unixfilesystem
└── pt1:passthru
    └── 27d8e028f4bfResource:unixfilesystem
demoResc:unixfilesystem
$ ils -L
/tempZone/home/rods:
  rods              0 def;pt1;27d8e028f4bfResource            5 2022-12-01.21:24 & foo
        generic    /var/lib/irods/Vault/home/rods/foo
  rods              0 def;pt0;110eb55edbccResource           10 2022-12-01.21:25 & goo
        generic    /var/lib/irods/Vault/home/rods/goo
  rods              1 def;pt1;27d8e028f4bfResource           10 2022-12-01.21:23 & goo
        generic    /var/lib/irods/Vault/home/rods/goo
```
I'm connected to the catalog service provider which is neither `27d8e028f4bf` or `110eb55edbcc`, so the vote would not be skewed.

Scenario 1:
 - Resource holds the only replica
 - Resource host iRODS server is paused
```
$ irods-grid pause --hosts=27d8e028f4bf
$ iget foo -
<hangs>
```
* If you resume the iRODS server via `irods-grid`, the `iget` will complete successfully.

Scenario 2:
 - Resource holds the only replica
 - Resource host iRODS server is stopped
```
$ irods-grid shutdown --hosts=27d8e028f4bf
$ iget foo -
remote addresses: 172.20.0.3 ERROR: getUtil: get error for - status = -305111 USER_SOCK_CONNECT_ERR, Connection refused
```

Voting does not seem to consider the status of the iRODS server for the host of a given resource in hierarchy resolution.


I found this a little surprising at first, but as I mentioned before, I think this is a reasonable approach. If the iRODS server being running or paused/shutdown was considered in the hierarchy resolution process, it would require redirects to every server hosting a replica to check the status. So, if a server is knowingly going to be shut down, the read/write weight for the `passthru` resource above the resource being hosted by the server in question should be set to 0 (equivalent to marking it "down") by the administrator.

Does this seem like a reasonable explanation? Is this behavior expected from your perspective, @kript?

**Comment by trel** [comment]
I think I agree.

Until an iRODS deployment can handle surprisingly-down machines more gracefully (most probably via #3480)... it seems the administrators should prepare for known-down/paused machines with policy change via read/write weights in the hierarchy.   This will be the most flexible way to have deterministic behavior during maintenance.

Note that 'marking down' is still problematic/inconsistent and its use should be considered deprecated (https://github.com/irods/irods/issues/4508#issuecomment-1009485419) as we work towards a richer set of passthru settings (in addition to read/write) (#4548, #4922, #6095)

**Comment by kript** [comment]
I agree with the arguments here, however I think there is a lot of complexity and potential for confusion around server availability.

Adding something like a summary of the two  responses here to the best practices guide around server availability (especially that marking down is effectively deprecated) would be helpful, as otherwise people might design an outage response based on incomplete information ('mark down and we're safe' or 'pause a server and we can reboot it' - both of which are true... mostly, under some conditions!).

**Comment by trel** [comment]
The control plane has been deprecated (https://github.com/irods/irods/issues/8479) and will be removed in iRODS 5.0.0 (https://github.com/irods/irods/issues/7229).

**Comment by korydraughn** [comment]
iRODS 5 is available. No additional work will be put into the control plane.

Closing.
